### PR TITLE
Fix: GasTiming: should only accept string types

### LINKED ISSUE #14764: [Non-Prod Bug]:  GasTiming: should only accept string types
### Describe the bug

The GasTiming component should only accept its `maxFeePerGas` and `maxPriorityFeePerGas` params as string types.  Instead, it is currently accepting both number and string types. 

It is confirmed that it should only be accepting a string type based on this @typedef in `metamask-extension/ui/hooks/gasFeeInput/useGasFeeInputs.js`. 
```
/**
 * @typedef {Object} GasFeeInputReturnType
 * @property {DecGweiString} [maxFeePerGas] - the maxFeePerGas input value.
 * @property {DecGweiString} [maxPriorityFeePerGas] - the maxPriorityFeePerGas
 *  input value.
```

When GasTiming receives a number value it produces this warning in development: 
image

Something observed was this value starts as a string and then turns into a number type.
Screenshot 2022-05-12 at 8 06 41 PM

### Steps to reproduce

1. Issue a transaction
2. Confirm GasTiming is working as expected
3. Observe the following errors shown in the screenshot above are not present with these changes

### Error messages or log output

_No response_

### Version

10.14.0

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

This bug does not appear to affect production.

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

It appears that `ui/components/app/gas-details-item/gas-details-item.js` is passing both number and string types to the GasTiming component. We should only be accepting the string type here. 

Audit of all instances of the GasTiming component:
- `ui/components/app/edit-gas-display/edit-gas-display.component.js`
- `ui/components/app/gas-details-item/gas-details-item.js`
- `ui/pages/confirm-transaction-base/confirm-transaction-base.component.js` 

## Description

Fixes https://github.com/MetaMask/metamask-extension/issues/14764

## Screenshots/Screencaps

Screenshot 2022-05-07 at 9 36 26 PM

## Manual Testing Steps

1. Issue a transaction
2. Confirm GasTiming is working as expected 
3. Observe the following errors shown in the screenshot above are not present with these changes

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
Code looks good to me! But I don't have context on what types this component should take so will remove myself as a reviewer and add @darkwing as git blame is suggesting he was last to touch this prop and may have the context required to answer the question above.

**Comment by darkwing** [comment]
@digiwand Are these arguments numbers being passed in as strings because we're referencing an `input` tag's `value`?  Could we use `valueAsNumber` (https://davidwalsh.name/input-valueasnumber)?

**Comment by digiwand** [comment]
@darkwing First and foremost, thank you for sending this gem over! This is my first time hearing about `valueAsNumber`. Secondly, thank you for the nudge for me to double-check this! 

It turns out that we should be expecting a string value based on this @typedef in `metamask-extension/ui/hooks/gasFeeInput/useGasFeeInputs.js`. 
```
/**
 * @typedef {Object} GasFeeInputReturnType
 * @property {DecGweiString} [maxFeePerGas] - the maxFeePerGas input value.
 * @property {DecGweiString} [maxPriorityFeePerGas] - the maxPriorityFeePerGas
 *  input value.
```

Something I found interesting was this value starts as a string and then turns into a number type.
Screenshot 2022-05-12 at 8 06 41 PM

I put this PR in draft mode to dig deeper into the fix. Instead of accepting both number and string type, GasTiming should only be accepting the string type for `maxFeePerGas` and `maxPriorityFeePerGas`

**Additional:** 
I also took some more time to look through how these gas values are set for the GasTiming component. I don't know if anyone else can make sense of it or find any kind of use for it, but I will share it just in case.https://app.diagrams.net/#G106C79uiq4CuYYKVRcph37DF9nz8XmOLm

TLDR; the `maxFeePerGas` and `maxPriorityFeePerGas` values for each GasTiming component instance comes from `useGasFeeInputs`. More investigation needed into `useGasFeeInputs`

**Comment by darkwing** [comment]
@digiwand Is this fixed by thee PropTypes PR?  If so, can we close this?

**Comment by digiwand** [comment]
Yes! Thank you @darkwing
